tofu/geom/_core.py

```diff
@@ -436,19 +436,23 @@ def _checkformat_inputs_dgeom(
             Poly = Poly.T
 
         # Elimininate any double identical point
-        ind = np.sum(np.diff(Poly, axis=1) ** 2, axis=0) < 1.0e-12
+        ind = np.sum(np.diff(np.concatenate((Poly, Poly[:, 0:1]), axis=1),
```

Comment:
why do you close the poly ?

Reply:
We want a bool array ind the same shape as Poly, to tell us which successive points are redundant.
So:
* We use np.diff() => ind will be of shape (npts-), if poly has shape (2, npts). Concatenating the first poitn at the end (i.e. closing the polygon) will yield the good shape for ind
* If we don't close the poly here, and if the first point and the last point are identical (poly already closed), np.diff() will not detect it. So closing the poly in np.diff() is actually a good way of making sure we remove (at line 443) the last point if the poly was already closed